app/controllers/courses_controller.rb

```diff
@@ -229,6 +229,63 @@ def users
             end
   end
 
+  action_auth_level :add_users_from_emails, :instructor
+  def add_users_from_emails
```

Comment:
@michellexliu Since you worked with roster upload, before I continue further, could you take a quick look at the logic below?